kolibri/core/content/management/commands/importcontent.py

```diff
@@ -197,22 +197,25 @@ def _start_file_transfer(self, f, filetransfer, overall_progress_update):
             original_progress = self.progresstrackers[0].get_progress()
 
             with filetransfer, self.start_progress(total=filetransfer.total_size) as file_dl_progress_update:
-                # If size of the source file is smaller than the the size
-                # indicated in the database, it's very likely that the source
-                # file is corrupted. Skip this file.
-                if filetransfer.total_size < f.file_size:
-                    e = "File {} is corrupted.".format(filetransfer.source)
-                    logger.error("An error occurred during content import: {}".format(e))
-                    overall_progress_update(f.file_size)
-                    return True, 1
-
                 for chunk in filetransfer:
                     if self.is_cancelled():
                         filetransfer.cancel()
                         return True, 0
                     length = len(chunk)
                     overall_progress_update(length)
                     file_dl_progress_update(length)
+
+                # If size of the destination file is smaller than the size
+                # indicated in the database, it's very likely that the source
+                # file is corrupted. Skip importing this file.
+                dest_file_size_on_disk = os.path.getsize(filetransfer.dest)
+                if dest_file_size_on_disk < f.file_size:
+                    e = "File {} is corrupted.".format(filetransfer.source)
+                    logger.error("An error occurred during content import: {}".format(e))
+                    os.remove(filetransfer.dest)
```

Comment:
Could we add an additional test to guarantee the cleanup behaviour also?

Reply:
Yes I will do that!

Reply:
Sounds good!